packages/mongodb-chatbot-server/src/routes/responses/createResponse.ts

```diff
@@ -254,15 +301,23 @@ interface LoadConversationByMessageIdParams {
   messageId?: string;
   conversations: ConversationsService;
   headers: Record<string, string>;
+  metadata?: Record<string, string>;
+  userId?: string;
+  store: boolean;
 }
 
-async function loadConversationByMessageId({
+const loadConversationByMessageId = async ({
   messageId,
   conversations,
   headers,
-}: LoadConversationByMessageIdParams): Promise<Conversation> {
+  metadata,
+  userId,
+  store,
+}: LoadConversationByMessageIdParams): Promise<Conversation> => {
   if (!messageId) {
-    return await conversations.create();
+    return await conversations.create({
+      customData: { metadata, userId, store },
+    });
```

Comment:
see note in tests about making userId and store top-level optional properties on the Conversation object. pls update this (and any other relevant places) accordingly.